common/rfb/SSecurityPlain.cxx

```diff
@@ -53,6 +55,13 @@ bool PasswordValidator::validUser(const char* username)
   for (size_t i = 0; i < users.size(); i++) {
     if (users[i] == "*")
       return true;
+#if !defined(WIN32) && !defined(__APPLE__)
+    if (users[i] == "%u") {
+      struct passwd *pw = getpwuid(geteuid());
```

Comment:
Why effective uid rather than real uid? I would assume most other things use the real uid, so this difference might be surprising.

Reply:
Oh yes, that was a mistake. It must be getuid or else it will be necessary to log in as root when Xvnc is installed with the set-user-id bit on.